### DISCUSSION TITLE
getServerSideProps error

### DISCUSSION THREAD

**Discussion by marshallshelly** [vercel/next.js#13940] [Help] [answered]
# Bug report

## Describe the bug

I fetched a list of hotels from an API endpoint and loaded them into the site. I also created dynamic routes for each hotel. Except for one hotel, every other hotel works (i.e their routes). When I click on the other hotels, it takes me to their respective route using `[slug].js`.

This is the error I got
![error](https://user-images.githubusercontent.com/23221413/84156984-caec1d80-aa87-11ea-9da1-e2087c52e014.PNG)

This is how I dynamically routed them
```
<Link
            href={`/${linkType}/[slug]`} //linkType == hotels
            as={`/${linkType}/${kebabCase(title)}`} // kebabCase from lodash
          >
            <a className="text-primary">Read more</a>
</Link>
```
This is how I used `getServerSideProps(context)` on `[slug].js`
```
export async function getServerSideProps(context) {
  let { slug } = context.query;
  let changeCase = startCase(slug); // startCase from lodash
  const res = await fetch(
    `${process.env.API_URL}/hotels?hotelName=${changeCase}`
  );
  const data = await res.json();
  return {
    props: { hotel: data[0] },
  };
}
```
## Expected behavior

![expected](https://user-images.githubusercontent.com/23221413/84157655-9e84d100-aa88-11ea-90c2-19f3510cafad.PNG)

## Screenshots
![hotels](https://user-images.githubusercontent.com/23221413/84158065-0e935700-aa89-11ea-91e2-133b312a60e2.PNG)

## System information

- OS: Windows 10 64bit
- Browser Google Chrome
- Version of Next.js: 9.4.4
- Version of Node.js: v12.18.0

## Additional context

I am using strapi as backend.
Only one hotel not working. Rest of them are working fine.

**Comment by zeekrey** [comment, marked as answer]
As the error shows, data[0] is undefined. Maybe you could check what exactly is returned by 

```
const res = await fetch(
    `${process.env.API_URL}/hotels?hotelName=${changeCase}`
  );
```

**Reply by marshallshelly** [reply]
Thank you @zeekrey for the help. I checked if `data[0]` was returning anything but unfortunately it wasn't.

The reason was
`hotelName: Hotel VPN Residency`
but what I was sending was
`hotelName: Hotel Vpn Residency`

The case difference caused the issue.

I fixed it by adding `_contains` as mentioned in Strapi [documentation](https://strapi.io/documentation/v3.x/content-api/parameters.html#filters)
```
const res = await fetch(
    `${process.env.API_URL}/hotels?hotelName_contains=${changeCase}`
  );
```
